app/src/main/java/org/stacks/app/ui/homepage/HomePageViewModel.kt

```diff
@@ -1,12 +1,63 @@
 package org.stacks.app.ui.homepage
 
 import androidx.hilt.lifecycle.ViewModelInject
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.flow.*
+import org.blockstack.android.sdk.Blockstack
 import org.stacks.app.data.interfaces.IdentityRepository
 import org.stacks.app.ui.BaseViewModel
+import org.stacks.app.ui.homepage.HomePageViewModel.UserAuthState.Authenticated
+import org.stacks.app.ui.homepage.HomePageViewModel.UserAuthState.Unauthenticated
+import timber.log.Timber
 
 class HomePageViewModel
 @ViewModelInject constructor(
-    private val identityRepository: IdentityRepository
+    identityRepository: IdentityRepository,
+    blockstack: Blockstack
 ) : BaseViewModel() {
 
+    // Outputs
+    private val authenticatedState = MutableStateFlow(Authenticated)
+    private val userAvatarImageUrl = BroadcastChannel<String?>(1)
+
+    init {
+        identityRepository
+            .observe()
+            .onEach {
+                authenticatedState.emit(
+                    if (it.isEmpty()) {
+                        Unauthenticated
+                    } else {
+                        Authenticated
```

Comment:
I fell there could be a domain use case in between the repo and the viewmodel, that would return the authentication state, with the first identity on it, for example.